src/subscription_manager/cli_command/abstract_syspurpose.py

```diff
@@ -173,8 +173,9 @@ def _is_provided_value_valid(self, value):
         # First check if the the value is in the valid_fields
         ret = False
         valid_fields = self._get_valid_fields()
+
         if self.attr in valid_fields:
-            if value in valid_fields[self.attr]:
+            if value.lower() in [x.lower() for x in valid_fields[self.attr]]:
```

Comment:
This is a good starting point :) We can do a bit better: instead of building a temporary list with the lowercase values, we can directly compare the existing elements in the list, and return as soon as a match is found. This is doable with the builtin [any()](https://docs.python.org/2.7/library/functions.html#any) function.

Another point is regarding the string comparison: lowercasing the strings works reliably as a case-insensitive comparison only for ASCII strings. There are languages (e.g. Turkish, Greek) where in some cases there is no 1:1 relation between an uppercase and a lowercase character. Because of this, a better way is to use [str.casefold()](https://docs.python.org/3.5/library/stdtypes.html#str.casefold).